src/services/resource-creator.js

```diff
@@ -5,55 +5,71 @@ const { ErrorHTTP422 } = require('./errors');
 const ResourceGetter = require('./resource-getter');
 const CompositeKeysManager = require('./composite-keys-manager');
 
-function ResourceCreator(model, params) {
-  const schema = Interface.Schemas.schemas[model.name];
-
-  this.perform = function perform() {
-    const promises = [];
-    const recordCreated = model.build(params);
-
-    if (model.associations) {
-      _.forOwn(model.associations, (association, name) => {
-        if (association.associationType === 'BelongsTo') {
-          promises.push(recordCreated[`set${_.upperFirst(name)}`](params[name], { save: false }));
-        }
-      });
+const getSetter = (associationType, name) => {
+  let setter;
+  if (associationType === 'HasOne') {
+    setter = `set${_.upperFirst(name)}`;
+  } else if (['BelongsToMany', 'HasMany'].includes(associationType)) {
+    setter = `add${_.upperFirst(name)}`;
+  }
+  return setter;
+};
+
+const getPromisesBeforeSave = (record, params) => (promises, [name, association]) => {
+  if (association.associationType === 'BelongsTo') {
+    promises.push(record[`set${_.upperFirst(name)}`](params[name], { save: false }));
+  }
+  return promises;
+};
+
+const getPromisesAfterSave = (record, params) => (promises, [name, association]) => {
+  const setter = getSetter(association.associationType, name);
+  if (setter) {
+    promises.push(record[setter](params[name]));
+  }
+  return promises;
+};
+
+class ResourceCreator {
+  constructor(model, params) {
+    this.model = model;
+    this.params = params;
+    this.schema = Interface.Schemas.schemas[model.name];
+  }
+
+  async handleSave(record, callback) {
+    const { associations } = this.model;
+    if (associations) {
+      const promisesBeforeSave = Object.entries(associations)
+        .reduce(callback(record, this.params), []);
+      await P.all(promisesBeforeSave);
+    }
+  }
+
+  async perform() {
```

Comment:
The create operation should be a very readable and simple list of invocation for example:

perform:
  // buildInstance
  // handleAssociationsBeforeSave
  // saveInstance (validate then save)
  // handleAssociationsAfterSave
  // appendCompositePrimary
  // return makeResourceGetter()